### PR TITLE
Customize style of the results list

### PR DESCRIPTION
**Pull request by Antonio-Maranhao**
## Overview

Adds a new menu to customize the style of the results list.

 - Truncated/Show full values
 - Font size: small, medium, large

image

## Testing recommendations

 - Go to documents list and switch to the Metadata view
 - Click the new button to show the menu
 - Click on `Show full values`
   - The full text value is displayed for each row
   - Menu label changes to 'Truncate values'
 - Click on `Small`, `Medium`, `Large` and verify the font size is updated
 - Switch to the Table view, and verify the same options also work
 - Switch to the JSON view, the menu should only show the 'Font size' options

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;

### CODE REVIEW

**Comment by wohali** [comment]
Love the idea, not crazy about the verbose checkbox for "Truncate values."

I'd suggest inverting the sense of the checkbox, making it "Expand values" instead?

**Comment by Antonio-Maranhao** [comment]
@wohali @popojargo I had some time to rework this. What do you think?

"The good, old Fauxton that we know and love..."
image

-"But it'd be great to see the full values"
-"Sure thing"
image

-"Too much scrolling... MORE DENSITY"
-"Okie dokie"
image

-"Argh... I need new glasses. Magnify, pls"
-"aye aye captain"
image

**Comment by popojargo** [comment]
Wow looks good! 👍

**Comment by Antonio-Maranhao** [comment]
@popojargo I pushed all the changes, could you pls review it?

File: app/addons/documents/index-results/reducers.js [hunk 1]
```diff
@@ -61,9 +62,34 @@ const initialState = {
...
   switch (action.type) {
 
+    case ActionTypes.INDEX_RESULTS_SET_STYLE:
+      const newStyle = {
+        ...state.resultsStyle,
+        ...action.resultsStyle
+      };
+      storeStyle(newStyle);
+      return Object.assign({}, state, {
+        resultsStyle: newStyle
```
Comment (popojargo) [inline app/addons/documents/index-results/reducers.js]: Why don't you use the spread operator here as well?

File: app/addons/documents/index-results/containers/IndexResultsContainer.js [hunk 2]
```diff
@@ -56,7 +58,8 @@ const mapStateToProps = ({indexResults}, ownProps) => {
     docType: getDocType(indexResults),
     fetchParams: getFetchParams(indexResults),
     queryOptionsParams: getQueryOptionsParams(indexResults),
-    partitionKey: ownProps.partitionKey
+    partitionKey: ownProps.partitionKey,
```
Comment (popojargo) [inline app/addons/documents/index-results/containers/IndexResultsContainer.js]: This should be in another PR right?

File: app/addons/documents/components/results-toolbar.js [hunk 3]
```diff
@@ -83,5 +97,7 @@ ResultsToolBar.propTypes = {
   isLoading: PropTypes.bool.isRequired,
   hasResults: PropTypes.bool.isRequired,
   isListDeletable: PropTypes.bool,
-  partitionKey: PropTypes.string
+  partitionKey: PropTypes.string,
```
Comment (popojargo) [inline app/addons/documents/components/results-toolbar.js]: Unrelated?

File: app/addons/documents/components/results-toolbar.js [hunk 4]
```diff
@@ -13,12 +13,18 @@ import PropTypes from 'prop-types';
...
 import BulkDocumentHeaderController from "../header/header";
+import ResultsOptions from './results-options';
 import Components from "../../components/react-components";
+import Constants from '../constants';
 import Helpers from '../helpers';
 
 const {BulkActionComponent} = Components;
 
 export class ResultsToolBar extends React.Component {
+  constructor (props) {
```
Comment (popojargo) [inline app/addons/documents/components/results-toolbar.js]: Useless constructor

File: app/addons/documents/components/results-options.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      fontSize: size
+    });
+  }
+
+  getDensitySection() {
+    const densityItems = [{
+      title: 'Show full values',
+      onClick: this.toggleTextOverflow
+    }];
+    if (this.props.resultsStyle.textOverflow === Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_FULL) {
```
Comment (popojargo) [inline app/addons/documents/components/results-options.js]: I think you should define the title before creating the densityItem. It's more error prone if you mutate the array at a certain index after its creation.

File: app/addons/documents/assets/less/index-results.less [hunk 6]
```diff
@@ -34,6 +34,39 @@
...
   }
+
+  .text-overflow-switch {
+    min-width: 145px;
+    min-height: 26px;
+    padding: 8px 0;
+  }
+
+  .toolbar-dropdown {    
+    // padding-top: 10px;
```
Comment (popojargo) [inline app/addons/documents/assets/less/index-results.less]: I guess we can remove those comments

File: app/addons/documents/__tests__/table-row.test.js [hunk 7]
```diff
@@ -54,4 +56,34 @@ describe('Docs Table Row', () => {
...
+      docChecked={sinon.stub()}
+      el={elem}
+      data={data}
+      index={0}
+      docIdentifier={elem.id}
+      isSelected={false}
+      textOverflow={Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_TRUNCATED}
+    />);
+
+    expect(wrapper.find('td.showall').exists()).toBe(false);
```
Comment (popojargo) [inline app/addons/documents/__tests__/table-row.test.js]: can this be replaced by `expect(wrapper.find('td.showall')).to.not.exist;` ?

**Comment by Antonio-Maranhao** [comment]
Thanks @popojargo. 
There's no special config. The button should always be visible from All Docs or any view. Do you see any errors in the console?

**Comment by popojargo** [comment]
@Antonio-Maranhao My bad, I had a problem with my local setup.

The icon doesn't look centered (there's a tiny margin): https://puu.sh/BXtqQ/d938cf448b.gif 

Also, I find it a bit weird that the button keeps moving from right to left depending on what's the view.

Ex: https://puu.sh/BXtnP/a251f74de6.png

The feature works fine on Chrome, Edge and IE11 just in case you're wondering ^^

File: app/addons/documents/__tests__/table-row.test.js (hunk 7 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/__tests__/table-row.test.js]: No because `find()` will always return a Wrapper even when there is no match. You need to either call `exists()` or check if `length` is zero.

File: app/addons/documents/components/results-toolbar.js (hunk 4 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/components/results-toolbar.js]: Removed

File: app/addons/documents/components/results-options.js (hunk 5 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/components/results-options.js]: Agree... it's changed.

File: app/addons/documents/assets/less/index-results.less (hunk 6 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/assets/less/index-results.less]: Removed

File: app/addons/documents/components/results-toolbar.js (hunk 3 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/components/results-toolbar.js]: Yes, the `partitionKey` was introduced in another PR

File: app/addons/documents/index-results/containers/IndexResultsContainer.js (hunk 2 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/index-results/containers/IndexResultsContainer.js]: It was added in another PR. It's only highlighted because of the added comma at the end of the line.

File: app/addons/documents/index-results/reducers.js (hunk 1 above)
Comment (Antonio-Maranhao) [reply app/addons/documents/index-results/reducers.js]: Updated... using the spread op in all actions
